bundle/src/main/java/com/adobe/acs/commons/httpcache/engine/impl/HttpCacheEngineImpl.java

```diff
@@ -623,4 +620,23 @@ public TabularData getRegisteredPersistenceStores() throws OpenDataException {
 
         return tabularData;
     }
+
+    /**
+     * Serves the cache content into the sling response.
+     * Has a fallback for wrappers that do not support the outputstream.
+     * @param response
+     * @param cacheContent
+     * @throws IOException
+     */
+    private void serveCacheContentIntoResponse(SlingHttpServletResponse response, CacheContent cacheContent)
+            throws IOException
+    {
+        try{
+            IOUtils.copy(cacheContent.getInputDataStream(), response.getOutputStream());
```

Comment:
Perhaps it would be better to check the `response.getContentType() => text/html` to see if the writer() can be used?

Feels weird using the exception to control flow of code.

@justinedelson thoughts on the best/definitive check for which to use?

Reply:
Can we be 100% sure that if the response type is text/html, the print writer is supported?
If so I guess that's indeed the best.

Reply:
I don't see why this would be specific to the case where the response type is `text/html`. To @davidjgonzalez 's question, although I agree in general that exceptions shouldn't be used for control flow, in this case I don't really see an alternative. The state of the response object (i.e. whether the output stream or writer has been returned) is not available.